src/runtime/chan.go

```diff
@@ -37,11 +37,78 @@ func chanDebug(ch *channel) {
 	}
 }
 
+// blockchain is a chain of blocked operations
+type blockchain struct {
+	// next blocked operation
+	next *blockchain
+
+	// t is the task to wake up
+	// the pointer state holds the data buffer unless this is a select
+	// the data field should be set to zero if this is an rx op when the channel closes
```

Comment:
I find this a bit confusing.

> the pointer state holds the data buffer unless this is a select

What does the pointer state contain when it is a select?

> the data field should be set to zero if this is an rx op when the channel closes

Why is this? I'd guess it is so that the GC can collect that object, but it would avoid some guesswork by a reader to spell it out explicitly.

Also, for slightly longer form comments I would personally prefer full sentences, starting with a capital letter and ending in a dot. Related (but not the same thing): https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences

Reply:
>I'd guess it is so that the GC can collect that object

It is not. The data field holds the comma-ok value, and setting to 0 marks it as false.